department-of-veterans-affairs/caseflow
# Enable screenreader to read success alerts

Resolves #14272

### Description
When we originally created the `Alert` component in #498, we filtered our alerts so that only error alerts were given the accessibility `role=alert` attribute for screenreader recognition.  This PR removes that filtering and places the `role=alert` attribute directly on the alert div so that both errors and success alerts will be properly recognized by screenreader and therefore 508 accessible.

Reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role

### Acceptance Criteria
- [ ] Screenreader recognizes and reads both error and success alerts

### Testing Plan
1. Enable screenreader (command + F5 on Mac)
2. Choose an attorney with cases assigned (e.g. BVASCASPER1)
3. Run through the attorney decision drafting process and ensure the success alert is correctly read by screenreader upon completion 
4. Choose a judge with cases to review (e.g. BVAAABSHIRE)
5. Run through the judge decision review process and ensure the success alert is correctly read by screenreader upon completion
6. Begin the judge decision review process but do not select any Special Issues to ensure the error alert is correctly read by screenreader

### User Facing Changes

#### Success alert recognized on attorney decision draft

<img width="1358" alt="Screen Shot 2020-06-10 at 5 18 58 PM" src="https://user-images.githubusercontent.com/18618189/84408010-99836700-abd9-11ea-9e26-4d2fd58d8ae3.png">


#### Success alert recognized on judge decision review

<img width="1376" alt="Screen Shot 2020-06-11 at 11 27 01 AM" src="https://user-images.githubusercontent.com/18618189/84407981-8ff9ff00-abd9-11ea-92bb-d56d21cb135a.png">

#### Error alert recognized when no Special Issues are selected

<img width="1371" alt="Screen Shot 2020-06-11 at 11 29 29 AM" src="https://user-images.githubusercontent.com/18618189/84408495-334b1400-abda-11ea-9a54-b890a27b4efc.png">
